sdks/python/container/boot.go

```diff
@@ -105,18 +107,57 @@ func main() {
 
 	// (2) Retrieve and install the staged packages.
 
-	dir := filepath.Join(*semiPersistDir, "staged")
+	func() {
 
-	files, err := artifact.Materialize(ctx, *artifactEndpoint, info.GetRetrievalToken(), dir)
-	if err != nil {
-		log.Fatalf("Failed to retrieve staged files: %v", err)
-	}
+        installCompleteFile := filepath.Join(os.TempDir(), "beam.install.complete")
 
-	// TODO(herohde): the packages to install should be specified explicitly. It
-	// would also be possible to install the SDK in the Dockerfile.
-	if setupErr := installSetupPackages(files, dir); setupErr != nil {
-		log.Fatalf("Failed to install required packages: %v", setupErr)
-	}
+		// skip if install already complete
+		_, err = os.Stat(installCompleteFile)
+		if err == nil {
+			return
```

Comment:
I think this should live within the locked section.

Reply:
There is no need to enter the locked section when the install was already completed (hence the comment above).

Reply:
Is this check really necessary? In case of a race condition when multiple workers complete this section before the install file has been created, only the later check will catch that. For simplicity, I'd remove this check entirely, unless the locking is very expensive.